webapp/src/main/webapp/openmrs.js

```diff
@@ -296,7 +296,8 @@ function gotoUser(select, userId) {
 		day = birthdate.substr(0, 2); 
 	} 
 	
-	var localeBirthDate = new Date(); 
+	var localeBirthDate = new Date();
+	localeBirthDate.setDate(1); // This is necessary to avoid this bug:  TRUNK-4035
```

Comment:
All months have a day 1, so setting a month and year when the day is set to 1 has no issues.  But not all months have a day 31.  So if you instantiate a new Date() on the 31st of the month, it will have a "day" component of 31.  If you then call the "setMonth()" method to change the month to February, you end up in a situation where the date would be February 31st, which is illegal.  Rather than fail, JS seems to just treat this as March 3.  So although we set the month to February, it _actually_ ends up getting set to March.  By first setting the day to 1, it avoids this possibility.  @dkayiwa

Reply:
Thanks for the very clear explanation! 👍